Shopify/cli
# Add plugin to deduplicate React dependency in extensions

### WHY are these changes introduced?

Fixes a bug where having multiple _copies_ (not versions) of React in the dependency tree causes extensions to fail.



### WHAT is this pull request doing?



Adds an ESBuild plugin to ensure that React requests are resolved to a single copy, per extension. Some prior art/guidance:
- https://github.com/evanw/esbuild/issues/2093#issuecomment-1066147618
- https://vitejs.dev/config/shared-options.html#resolve-dedupe

This feature can be deactivated (though you shouldn't need to) by setting `SHOPIFY_CLI_SKIP_ESBUILD_REACT_DEDUPLICATION` to something truthy.

### How to test your changes?



1. Use React 18 in a new embedded web app
2. Generate a checkout extension
3. Confirm that the checkout extension can be previewed

Without this fix, the checkout extension can't be seen and there's a pretty obscure error in the browser console.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
